### ISSUE TITLE
EntitySet deepcopy loses DataFrame `make_index` value.

### ISSUE THREAD

**Issue by tamargrey** [alteryx/featuretools#1483]
If a DataFrame in an EntitySet had its index created with `make_index`, that value gets stored on the DataFrame Accessor under the `make_index` attribute. 

However, because this attribute does not exist on the schema, it does not get copied in the deepcopy operation, and then when the schema gets passed to Woodwork init with `new_df.ww.init(schema=schema)`, we end up losing the `make_index` value, because that parameter will be ignored by the Woodwork init.

The reason on Woodwork's end to not allow passing a `make_index` parameter when initializing with a schema was that the DataFrame would not have actually had an index made through that `init` process, so it would be incorrect to indicate that it had.

However, in copying a woodwork DataFrame, we want to be able to pass along that `make_index` parameter, because a DataFrame that has had an index created for it should have `make_index=True` whether or not it was created from a copy of another DataFrame.

Proposed solution to `__deepcopy__` - set the make index value after Woodwork is initialized:

```python
if k == 'dataframe_dict':
    for df_name, new_df in copied_attr.items():
        schema = copy.deepcopy(self.dataframe_dict[df_name].ww.schema, memo=memo)
        new_df.ww.init(schema=schema)
        new_df.ww.make_index = self.dataframe_dict[df_name].ww.make_index
```

**Comment by thehomebrewnerd** [comment]
The changes implemented in PR #1507 should fix this issue. If an index is created a `created_index` key/value pair will be added to the Woodwork table metadata. This metadata will get copied as part of the deepcopy process, retaining the fact that the dataframe originally had an index column made.

Once #1507 is closed, we should verify that the metadata value is carried over properly:

```python
import pandas as pd
import featuretools as ft

es = ft.EntitySet()
df = pd.DataFrame({'data': ['a', 'b', 'c']})
es.add_dataframe(dataframe_name="test_df", dataframe=df, make_index=True, index='id')
assert 'created_index' in es['test_df'].ww.metadata
assert es['test_df'].ww.metadata['created_index'] == 'id'

import copy
es2 = copy.deepcopy(es)
assert 'created_index' in es2['test_df'].ww.metadata
assert es2['test_df'].ww.metadata['created_index'] == 'id'
es2.__eq__(es, deep=True)
```

Assuming no issues are found, we should be able to close this issue at that time.

**Comment by thehomebrewnerd** [comment]
Confirmed via the code above that the `created_index` key/value pair is retained in the dataframe metadata after a deepcopy operation. With the current code in the `woodwork-integration` branch, no further changes should be needed to resolve this issue.
